### PR TITLE
Add payment endpoints

### PR DESCRIPTION
**Pull request by GeekaN2**
Add payment endpoints

### CODE REVIEW

**Review by talyguryn — APPROVED** [review]

File: src/cloudpayments/index.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+const cloudpayments = new Cloudpayments();
+
+router.all('/check', cloudpayments.check);
+router.all('/pay', cloudpayments.pay);
+router.all('/fail', cloudpayments.fail);
+router.all('/recurrent', cloudpayments.recurrent);
+
+export default router;
```
Comment (nikmel2803) [inline src/cloudpayments/index.ts:69]: что если инитить роутер в конструкторе и возвращать 
`export default cloudpayments.router;`

File: src/cloudpayments/index.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import express from 'express';
```
Comment (neSpecc) [inline src/cloudpayments/index.ts:1]: the folder name should be more abstract

File: src/cloudpayments/index.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+const router = express.Router();
+
+/**
+ * Class for describing the logic of payment routes
+ */
+class Cloudpayments {
+  /**
+   * Route for verifying a user's payment
```
Comment (neSpecc) [inline src/cloudpayments/index.ts:23]: what does it mean 'verify' in this case?

File: src/cloudpayments/index.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+   * https://developers.cloudpayments.ru/#check
+   *
+   * @param req - cloudpayments request with payment details
+   * @param res - check result code
+   */
+  public async check(req: express.Request, res: express.Response): Promise<void> {
+    res.send({
+      code: 0, // Payment can be made
```
Comment (neSpecc) [inline src/cloudpayments/index.ts:31]: magic number

**Review by neSpecc — CHANGES_REQUESTED** [review]

File: src/cloudpayments/index.ts (hunk 3 above)
Comment (GeekaN2) [reply src/cloudpayments/index.ts:23]: verify : to confirm the truth or correctness of something.

Comment (neSpecc) [reply src/cloudpayments/index.ts:23]: put it in a jsdoc

**Review by nikmel2803 — APPROVED** [review]

File: src/billing/index.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import Cloudpayments from './cloudpayments';
+
+const cloudpayments = new Cloudpayments();
+
+export default cloudpayments.router;
```
Comment (neSpecc) [inline src/billing/index.ts:5]:
```ts 
export default class Billing {
  constructor(app){
    this.providerWebhooks = new CloudPaymentsWebhooks();

   app.use('/billing', this.providerWebhooks.getRouter())
  }
}
```

File: src/index.ts [hunk 6]
```diff
@@ -71,10 +71,10 @@ class HawkAPI {
    */
   constructor() {
     this.app.use(express.json());
-    this.app.post('/billing', billing.notifyCallback);
     this.app.use('/uploads', express.static(`./${process.env.UPLOADS_DIR || 'uploads'}`));
     this.app.use('/static', express.static(`./static`));
     this.app.use('/voyager', voyagerMiddleware({ endpointUrl: '/graphql' }));
+    this.app.use('/cloudpayments', cloudpaymentsRouter);
```
Comment (neSpecc) [inline src/index.ts:77]: move to `createContext`


```ts

new Billing(this.app)

return {
 ...
 billing
}

```

**Review by neSpecc — CHANGES_REQUESTED** [review]

File: src/billing/cloudpayments.ts [hunk 7]
```diff
@@ -0,0 +1,74 @@
+import express from 'express';
+
+import { CheckCodes, CheckResponse, PayCodes, PayResponse, FailCodes, FailResponse, RecurrentResponse, RecurrentCodes } from './types';
+
+/**
+ * Class for describing the logic of payment routes
+ */
+export default class CloudPaymentsWebhooks {
+  /**
+   * Get router for payments
```
Comment (neSpecc) [inline src/billing/cloudpayments.ts:10]:
```suggestion
   * Returns router for payments
```

**Review by neSpecc — APPROVED** [review]

**Pull request merged**